src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -113,27 +114,78 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
                 copyCommandDict['checksum'] = "-K adler32"
 
         copyCommandDict['options'] = ' '.join(unknown)
-
         copyCommandDict['source'] = self.createFinalPFN(sourcePFN)
         copyCommandDict['destination'] = self.createFinalPFN(targetPFN)
 
         copyCommand = self.copyCommand % copyCommandDict
         result += copyCommand
 
+        logging.debug("Actual command which failed: %s", copyCommand)
+
         if _CheckExitCodeOption:
             result += """
             EXIT_STATUS=$?
             echo "gfal-copy exit status: $EXIT_STATUS"
             if [[ $EXIT_STATUS != 0 ]]; then
-               echo "ERROR: gfal-copy exited with $EXIT_STATUS"
-               echo "Cleaning up failed file:"
-               %s
+                echo "ERROR: gfal-copy exited with $EXIT_STATUS"
+                echo "Source PFN: {source}"
+                echo "Target PFN: {destination}"
+                echo "Cleaning up failed file: {remove_command}"
             fi
             exit $EXIT_STATUS
-            """ % self.createRemoveFileCommand(targetPFN)
+            """.format(
+                source=copyCommandDict['source'], 
+                destination=copyCommandDict['destination'], 
+                remove_command=self.createRemoveFileCommand(targetPFN)
+            )
+        
+        return result
+
+    def createDebuggingCommand(self, sourcePFN, targetPFN, options=None, checksums=None):
+        """
+        Report environment information about a failed gfal-copy command
+        """
+        result = "#!/bin/bash\n"
+
+        copyCommandDict = {'checksum': '', 'options': '', 'source': '', 'destination': ''}
+
+        useChecksum = (checksums is not None and 'adler32' in checksums and not self.stageIn)
+
+        if not options:
+            options = ''
+
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--nochecksum', action='store_true')
+        args, unknown = parser.parse_known_args(options.split())
+
+        if not args.nochecksum:
+            if useChecksum:
+                checksums['adler32'] = "%08x" % int(checksums['adler32'], 16)
+                copyCommandDict['checksum'] = "-K adler32:%s" % checksums['adler32']
+            else:
+                copyCommandDict['checksum'] = "-K adler32"
+
+        copyCommandDict['options'] = ' '.join(unknown)
+        copyCommandDict['source'] = self.createFinalPFN(sourcePFN)
+        copyCommandDict['destination'] = self.createFinalPFN(targetPFN)
+
+        copyCommand = self.copyCommand % copyCommandDict
+
+        result += """
+        echo "gfal-copy command which failed: {copy_command}"
+        echo "Source PFN: {source}"
+        echo "Target PFN: {destination}"
+        echo "Proxy and JOBSTARTDIR: {setup_info}"
```

Comment:
I think only printing the proxy location is not going to be enough for Stephan / SST.
In addition to the first description of the issue, this comment summarizes the rest:
https://github.com/dmwm/WMCore/issues/11731#issuecomment-2306788582